internal/restic/node_windows.go

```diff
@@ -407,38 +408,91 @@ func (node *Node) fillGenericAttributes(path string, fi os.FileInfo, stat *statT
 // checkAndStoreEASupport checks if the volume of the path supports extended attributes and stores the result in a map
 // If the result is already in the map, it returns the result from the map.
 func checkAndStoreEASupport(path string) (isEASupportedVolume bool, err error) {
-	// Check if it's an extended length path
-	if strings.HasPrefix(path, uncPathPrefix) {
-		// Convert \\?\UNC\ extended path to standard path to get the volume name correctly
-		path = `\\` + path[len(uncPathPrefix):]
-	} else if strings.HasPrefix(path, extendedPathPrefix) {
-		//Extended length path prefix needs to be trimmed to get the volume name correctly
-		path = path[len(extendedPathPrefix):]
-	} else if strings.HasPrefix(path, globalRootPrefix) {
-		// EAs are not supported for \\?\GLOBALROOT i.e. VSS snapshots
-		return false, nil
-	} else {
-		// Use the absolute path
-		path, err = filepath.Abs(path)
-		if err != nil {
-			return false, fmt.Errorf("failed to get absolute path: %w", err)
+	var volumeName string
+	volumeName, err = prepareVolumeName(path)
+	if err != nil {
+		return false, err
+	}
+
+	if volumeName != "" {
+		// First check if the manually prepared volume name is already in the map
+		eaSupportedValue, exists := eaSupportedVolumesMap.Load(volumeName)
+		if exists {
+			// Cache hit, immediately return the cached value
+			return eaSupportedValue.(bool), nil
+		}
+		// If not found, check if EA is supported with manually prepared volume name
+		isEASupportedVolume, err = fs.PathSupportsExtendedAttributes(volumeName + `\`)
+		// If the prepared volume name is not valid, we will fetch the actual volume name next.
+		if err != nil && !errors.Is(err, windows.DNS_ERROR_INVALID_NAME) {
+			debug.Log("Error checking if extended attributes are supported for prepared volume name %s: %v", volumeName, err)
+			// There can be multiple errors like path does not exist, bad network path, etc.
+			// We just gracefully disallow extended attributes for cases.
+			return false, nil
```

Comment:
I think it will be much less intrusive to gracefully handle the errors and simple skip extended attributes since this is a side functionality. The other parts of the code can handle the path and file existence related errors instead.
There can be multiple errors like path does not exist, bad network path, etc.

Reply:
I agree, extended attributes aren't that important on Windows. Skipping them if we cannot detect whether they are supported is fine for me.

> Maybe we should show Warnings?

Hmm, there's nothing a user can do about such warnings. A debug log entry should be enough for now.